cylc/flow/xtriggers/workflow_state.py

```
@@ -14,18 +14,115 @@
+            Alternate cylc-run directory, e.g. for another user.
+
+    Returns:
+        tuple: (satisfied, result)
+
+        satisfied:
+            True if ``satisfied`` else ``False``.
+        result:
+            Dict {workflow_id, task_id, task_selector, flow_num}.
+
+    """
+    poller = WorkflowPoller(
+        workflow_task_id,
+        offset,
+        flow_num,
+        alt_cylc_run_dir,
+        TASK_STATUS_SUCCEEDED,
+        is_output, is_message,
+        old_format=False,
+        condition=workflow_task_id,
+        max_polls=1,  # (for xtriggers the scheduler does the polling)
+        interval=0,  # irrelevant for 1 poll
+        args=[]
+    )
+    if asyncio.run(poller.poll()):
+        return (
+            True,
+            {
+                "workflow_id": poller.workflow_id,
+                "task_id": f"{poller.cycle}/{poller.task}",
+                "task_selector": poller.selector,
+                "flow_num": poller.flow_num
+            }
+        )
+    else:
+        return (
+            False,
+            {}
+        )
+
+
+def validate(args: Dict[str, Any]):
+    """Validate workflow_state xtrigger function args.
+
+    Arguments:
+        workflow_task_id:
+            full workflow//cycle/task[:selector]
+        offset:
+            must be a valid status
+        flow_num:
+            must be an integer
+        alt_cylc_run_dir:
+            must be a valid path
+
+    """
+    try:
+        tokens = tokenise(args["workflow_task_id"])
+    except KeyError:
+        raise WorkflowConfigError(
+            # TODO better message
+            "Full ID needed: workflow//cycle/task[:selector].")
+
+    if any(
+        tokens[token] is None
+        for token in ("workflow", "cycle", "task")
+    ):
+        raise WorkflowConfigError(
+            "Full ID needed: workflow//cycle/task[:selector].")
+
+    if not isinstance(args["flow_num"], int):
+        raise WorkflowConfigError("flow_num must be an integer.")
```

But `flow_num` can be `None` too?

---

Actually no, it is defaulting to 1 despite the misleading type hint. But it should default to None. Fixed.